simpeg/inverse_problem.py

```diff
@@ -96,7 +96,7 @@ def counter(self, value):
         self._counter = value
 
     @property
-    def dmisfit(self):
+    def dmisfit(self) -> ComboObjectiveFunction:
```

Comment:
```suggestion
    def dmisfit(self) -> BaseObjectiveFunction:
```

Reply:
Not here, since the setter is enforcing Combo

Reply:
Right! My bad. I still think these properties should be more general, but we shouldn't tackle it in this PR. Let's leave your type hints then.